src/unix/solarish/mod.rs

```diff
@@ -1915,10 +1915,16 @@ pub const SHUT_RD: ::c_int = 0;
 pub const SHUT_WR: ::c_int = 1;
 pub const SHUT_RDWR: ::c_int = 2;
 
-pub const LOCK_SH: ::c_int = 1;
-pub const LOCK_EX: ::c_int = 2;
-pub const LOCK_NB: ::c_int = 4;
-pub const LOCK_UN: ::c_int = 8;
+// These don't exist on Solaris, but do on OpenSolaris derivatives
+// e.g. https://illumos.org/man/3C/flock
+cfg_if! {
```

Comment:
If something is illumos specific, I suspect we ought to move it to `illumos.rs` rather than introduce more conditional compilation directives?

Reply:
I don't know how much Illumos diverges from Solaris.  If it's a matter of a few things being added to Illumos then I'd think some directives wouldn't hurt.  If it's a matter of things being changed wholesale or a lot of new features being added then, yeah, I'm inclined to agree that a new module is the way to go.

I've no strong feelings either way but I don't have an Illumos install to check against.

Reply:
The divergence only increases over time.  Oracle made a proprietary fork of OpenSolaris more than a decade ago, and we made an open one at the same time.  We've been diverging ever since.

Regardless, I believe the module file already exists, alongside this file.

Reply:
With those constants moved into `illumos.rs` it looks like I've 1.70.0 building successfully for Solaris 11.4 (x86+SPARC) except for clippy. 🎉